src/vs/workbench/contrib/terminal/browser/links/terminalLinkDetectorAdapter.ts

```diff
@@ -0,0 +1,110 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Emitter } from 'vs/base/common/event';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { localize } from 'vs/nls';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { ITerminalSimpleLink, ITerminalLinkDetector, TerminalLinkType } from 'vs/workbench/contrib/terminal/browser/links/links';
+import { TerminalLink } from 'vs/workbench/contrib/terminal/browser/links/terminalLink';
+import { XtermLinkMatcherHandler } from 'vs/workbench/contrib/terminal/browser/links/terminalLinkManager';
+import { IBufferLine, ILink, ILinkProvider, IViewportRange } from 'xterm';
+
+export interface IShowHoverEvent {
+	link: TerminalLink;
+	viewportRange: IViewportRange;
+	modifierDownCallback?: () => void;
+	modifierUpCallback?: () => void;
+}
+
+/**
+ * Wrap a link detector object so it can be used in xterm.js
+ */
+export class TerminalLinkDetectorAdapter extends Disposable implements ILinkProvider {
+	private _activeLinks: TerminalLink[] | undefined;
+
+	private readonly _onDidActivateLink = this._register(new Emitter<ITerminalSimpleLink>());
+	readonly onDidActivateLink = this._onDidActivateLink.event;
+	private readonly _onDidShowHover = this._register(new Emitter<IShowHoverEvent>());
+	readonly onDidShowHover = this._onDidShowHover.event;
+
+	constructor(
+		private readonly _detector: ITerminalLinkDetector,
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+	) {
+		super();
+	}
+
+	async provideLinks(bufferLineNumber: number, callback: (links: ILink[] | undefined) => void) {
+		this._activeLinks?.forEach(l => l.dispose());
+		this._activeLinks = await this._provideLinks(bufferLineNumber);
+		callback(this._activeLinks);
+	}
+
+	private async _provideLinks(bufferLineNumber: number): Promise<TerminalLink[]> {
+		// Dispose of all old links if new links are provides, links are only cached for the current line
+		const links: TerminalLink[] = [];
+
+		let startLine = bufferLineNumber - 1;
+		let endLine = startLine;
+
+		const lines: IBufferLine[] = [
+			this._detector.xterm.buffer.active.getLine(startLine)!
+		];
```

Comment:
```suggestion
		const lines: IBufferLine[] = [this._detector.xterm.buffer.active.getLine(startLine)!];
```

Reply:
Kind of prefer it this way, it's a small file so the extra lines don't matter too much